modules/appgw/variables.tf

```diff
@@ -1,169 +1,632 @@
+          "TLS_RSA_WITH_AES_128_GCM_SHA256", "TLS_RSA_WITH_AES_256_CBC_SHA", "TLS_RSA_WITH_AES_256_CBC_SHA256", "TLS_RSA_WITH_AES_256_GCM_SHA384"]
+        )) == 0
+    ]]))
+    error_message = <<-EOF
+    Possible values for `ssl_policy_cipher_suites` are TLS_DHE_DSS_WITH_3DES_EDE_CBC_SHA, TLS_DHE_DSS_WITH_AES_128_CBC_SHA,
+    TLS_DHE_DSS_WITH_AES_128_CBC_SHA256, TLS_DHE_DSS_WITH_AES_256_CBC_SHA, TLS_DHE_DSS_WITH_AES_256_CBC_SHA256, TLS_DHE_RSA_WITH_AES_128_CBC_SHA,
+    TLS_DHE_RSA_WITH_AES_128_GCM_SHA256, TLS_DHE_RSA_WITH_AES_256_CBC_SHA, TLS_DHE_RSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA,
+    TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA,
+    TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA384, TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384, TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA,
+    TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256, TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256, TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA,
+    TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA384, TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384, TLS_RSA_WITH_3DES_EDE_CBC_SHA,
+    TLS_RSA_WITH_AES_128_CBC_SHA, TLS_RSA_WITH_AES_128_CBC_SHA256, TLS_RSA_WITH_AES_128_GCM_SHA256, TLS_RSA_WITH_AES_256_CBC_SHA,
+    TLS_RSA_WITH_AES_256_CBC_SHA256 and TLS_RSA_WITH_AES_256_GCM_SHA384.
+    EOF
+  }
+  validation {
+    condition     = length(flatten([for _, ssl_profile in var.ssl_profiles : ssl_profile.name])) == length(distinct(flatten([for _, ssl_profile in var.ssl_profiles : ssl_profile.name])))
+    error_message = "The `name` property has to be unique among all SSL profiles."
+  }
 }
 
-variable "subnet_id" {
-  description = "An ID of a subnet that will host the Application Gateway. Keep in mind that this subnet can contain only AppGWs and only of the same type."
+variable "frontend_ip_configuration_name" {
+  description = "Frontend IP configuration name"
+  default     = "public_ipconfig"
+  nullable    = false
   type        = string
 }
 
-variable "vmseries_ips" {
-  description = "IP addresses of VMSeries' interfaces that will serve as backends for the Application Gateway."
-  default     = []
-  type        = list(string)
+variable "listeners" {
+  description = <<-EOF
+  A map of listeners for the Application Gateway.
+
+  Every listener contains attributes:
+  - `name`                     - (`string`, required) The name for this Frontend Port.
+  - `port`                     - (`string`, required) The port used for this Frontend Port.
+  - `protocol`                 - (`string`, optional) The Protocol to use for this HTTP Listener.
+  - `host_names`               - (`list`, optional) A list of Hostname(s) should be used for this HTTP Listener.
+                                 It allows special wildcard characters.
+  - `ssl_profile_name`         - (`string`, optional) The name of the associated SSL Profile which should be used for this HTTP Listener.
```

Comment:
here, and in some other places as well, line wrapping at 130 column

Reply:
fixed